### ISSUE TITLE
App crash when trying to print

### ISSUE THREAD

**Issue by jraoatlogic** [katzer/cordova-plugin-printer#28]
We're trying to use the 0.6.x branch to print some html, it works on emulator (4.4.2 API 19 Google APIs), but crashes the app on LG Flex (4.4.2) and N7 (4.4.4), the log is:
```
10-16 10:14:23.818: E/AndroidRuntime(6756): FATAL EXCEPTION: pool-1-thread-1
10-16 10:14:23.818: E/AndroidRuntime(6756): Process: <our app id>, PID: 6756
10-16 10:14:23.818: E/AndroidRuntime(6756): android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.SEND typ=text/html flg=0x1 pkg=com.google.android.apps.cloudprint (has clip) (has extras) }
10-16 10:14:23.818: E/AndroidRuntime(6756):     at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1632)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at android.app.Instrumentation.execStartActivity(Instrumentation.java:1424)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at android.app.Activity.startActivityForResult(Activity.java:3424)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at android.app.Activity.startActivityForResult(Activity.java:3385)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at org.apache.cordova.CordovaActivity.startActivityForResult(CordovaActivity.java:859)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at de.appplant.cordova.plugin.printer.Printer.printViaGoogleCloudPrintApp(Printer.java:198)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at de.appplant.cordova.plugin.printer.Printer.access$300(Printer.java:46)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at de.appplant.cordova.plugin.printer.Printer$2$1.onContentReady(Printer.java:133)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at de.appplant.cordova.plugin.printer.Printer$ContentClient$1.run(Printer.java:337)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
10-16 10:14:23.818: E/AndroidRuntime(6756):     at java.lang.Thread.run(Thread.java:841)
10-16 10:14:23.818: W/ActivityManager(503):   Force finishing activity <our app id>
10-16 10:14:23.838: D/CordovaActivity(6756): Paused the application!
10-16 10:14:23.838: D/CordovaWebView(6756): Handle the pause
```

We tried this using the latest code in google-cloud-print branch and 0.6.1, both produces this crash, but 0.6.0 and 0.7.0 works.

**Comment by jrao1** [comment]
@katzer Sorry, I missed the notification of your comment. I just tested this again on Nexus 5 (4.4.4) using 0.6.2-dev, the crash still appears. The phone has Could Print installed, both the factory version (0.8.10) and the latest version (1.11b) are tested, and both crash.

**Comment by RonCan** [comment]
Did you guys ever found a solution? Please post here it'll help a lot of people too or any other alternative you came to use.

**Comment by jrao1** [comment]
No, didn't find a solution...

**Comment by batesy87** [comment]
Having the same issue and error android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.SEND typ=text/html flg=0x1 pkg=com.google.android.apps.cloudprint (has clip) (has extras) }

any joy with this?

Google cloud print is installed
Android 5.1.1

**Comment by batesy87** [comment]
Further to above, here is what I'm doing
cordova.plugins.printer.isAvailable(
	function (isAvailable) {
		if (isAvailable) {
			cordova.plugins.printer.print('<h1>TEST</h1>', 'Document.html', function () {
				alert('printing finished or cancelled')
			});
		}
	}
);

**Comment by cassioscofield** [comment]
I`m having the same issue on Moto X2 / Android 5.1.
Is this fixed yet? Or is there any work-around this problem?

**Comment by marty-krishner** [comment]
same

**Comment by lobobobo** [comment]
same problem..

**Comment by bmwertman** [comment]
I'm having the same issue. Google cloud print 1.17b installed on OnePlus One running Android 5.0.2

**Comment by bmwertman** [comment]
I'm using this in an application with a (sails.js)[http://sailsjs.org/] backend. If I install from master it breaks the websocket functionality built into sails which I'm depending on. Likewise if I remove the plugin completely, websockets break. With the plugin installed using `cordova plugin add de.appplant.cordova.plugin.printer` my websockets work but the app crashes when the `print` function gets called.

**Comment by lobobobo** [comment]
My issue got resolved when i installedMaster plugin, earlier  i installed stable plugin which use to crash my app abruptly.
You will find the MASTER plugin below link ::
https://github.com/katzer/cordova-plugin-printer#send-content-to-a-printer

**Comment by rk-panchal** [comment]
i have same issue  like , when i go for printing a div content , printing  functionality execute well but  css and style  of my page and fonts is not available on printing content.


if any one found solution plz mention here.
Thanks

**Comment by katzer** [comment]
Guys I've released 0.7.2. I hope it does fixes any issues. Also the sample app does print HTML with CSS and it's working fine.

<img width="23.8%" src="https://github.com/katzer/cordova-plugin-printer/blob/example/images/print-ios.png"></img>
